packages/camera/camera_android/lib/src/android_camera.dart

```diff
@@ -19,13 +20,19 @@ const MethodChannel _channel =
 
 /// The Android implementation of [CameraPlatform] that uses method channels.
 class AndroidCamera extends CameraPlatform {
+  /// Creates a new [CameraPlatform] instance.
+  AndroidCamera([@visibleForTesting CameraApi? hostApi])
```

Comment:
This should be named, not positional.

As a general rule of thumb: if something is a public API (where changing from positional to named would be a breaking change), and you are not *absolutely certain* that there could never be more parameters added later, don't use optional positional parameters. They don't scale at all to orthogonal parameters—and in this case that's especially true because no actual client could ever pass anything but null for this parameter, so if there were a parameter added later that was actually useful, everyone would have to pass a pointless `null`.